src/sc2api/sc2_unit.cc

```diff
@@ -14,6 +14,16 @@ bool Unit::IsBuildFinished() const {
     return build_progress >= 1.0f;
 }
 
+Tags GetTagsFromUnits(const Units& units) {
+    Tags tags;
+    for (const auto& unit : units) {
+        if (!unit) continue;
+        tags.push_back(unit->tag);
```

Comment:
I believe there is no need in checking for nullptr. This is much better to crash as nullptr in units list means some weird error in program.

Also, please try this solution, it allows to use algorithms from standard library and could be faster:
https://stackoverflow.com/questions/59840478/using-back-inserter-to-create-vector-from-single-member-of-struct